File: Level2/타겟 넘버.py
answer = 0
def DFS(numbers,tot,target,idx):
    global answer
    n = len(numbers)
    
    if idx == n and tot == target:
        answer+=1
        return
    if idx == n:
        return

    DFS(numbers,tot+numbers[idx],target,idx+1)
    DFS(numbers,tot-numbers[idx],target,idx+1)

def solution(numbers, target):
    global answer
    DFS(numbers,0,target,0)
    return answer

File: Level2/타겟넘버.cpp
#include <string>
#include <vector>

using namespace std;

//전역변수로 카운트
int cnt = 0;

void DFS(vector<int>& numbers, int target, int level, int tot) {
    if (level == numbers.size()) {
        if (tot == target)
            cnt++;
        return;
    }
    DFS(numbers, target, level+1, tot + numbers[level]);
    DFS(numbers, target, level+1, tot - numbers[level]);
    return;
}

int solution(vector<int> numbers, int target) {
    int answer = 0;
    
    DFS(numbers,target,0,0);
    answer = cnt;
    
    return answer;
}
